src/keywords.rs

```diff
@@ -975,6 +975,16 @@ pub const RESERVED_FOR_COLUMN_ALIAS: &[Keyword] = &[
     Keyword::END,
 ];
 
+// Global list of reserved keywords alloweed after FROM.
+// Parser should call Dialect::get_reserved_keyword_after_from
+// to allow for each dialect to customize the list.
+pub const ALLOWED_KEYWORD_AFTER_FROM: &[Keyword] = &[
```

Comment:
```suggestion
pub const RESERVED_FOR_TABLE_FACTOR: &[Keyword] = &[
```
Thinking something like this to keep the naming somewhat similar to the existing lists?

If this changes the dialect method `get_reserved_keyword_after_from` would be updated to match the naming